src/bandersnatch/mirror.py

```diff
@@ -112,17 +113,32 @@ def webdir(self) -> Path:
     def todolist(self) -> Path:
         return self.homedir / "todo"
 
-    async def synchronize(self) -> Dict[str, Set[str]]:
+    async def synchronize(
+        self, specific_packages: Optional[List[str]] = None
+    ) -> Dict[str, Set[str]]:
         logger.info(f"Syncing with {self.master.url}.")
         self.now = datetime.datetime.utcnow()
         # Lets ensure we get a new dict each run
         # - others importing may not reset this like our main.py
         self.altered_packages = {}
 
-        await self.determine_packages_to_sync()
-        await self.sync_packages()
-        self.sync_index_page()
-        self.wrapup_successful_sync()
+        if specific_packages is None:
+            # Changelog-based synchronization
+            await self.determine_packages_to_sync()
+            await self.sync_packages()
+            self.sync_index_page()
+            self.wrapup_successful_sync()
+        else:
+            # Synchronize specific packages. This method doesn't update the self.statusfile
```

Comment:
```suggestion
            # Synchronize specific packages. This method doesn't update the
            # self.statusfile
```

The CI complained that the line is too long. 
```console
./src/bandersnatch/mirror.py:132:89: E501 line too long (91 > 88 characters)
```

By any chance are the pre-commit hooks not running before your commits? This should be something that is caught by the Flake8 pre-commit hook. No worries though, it's a trival thing, none of us are immune from making mistakes :)

Reply:
I didn't get notification from pre-commit hooks. I'll check if it's installed correctly.

Reply:
CI should definitely be running pre-commit via tox. Please run tox locally too if you can. Should catch all this.